test/orizzonte/utility/nothing.cpp

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2017 Vittorio Romeo
+// MIT License |  https://opensource.org/licenses/MIT
+// http://vittorioromeo.info | vittorio.romeo@outlook.com
+
+#include "../../test_utils.hpp"
+#include <orizzonte/utility/nothing.hpp>
+
+using namespace orizzonte::utility;
+
+SA_SAME_TYPE(void_to_nothing_t<int>, int);
+SA_SAME_TYPE(void_to_nothing_t<void>, nothing);
+SA_SAME_TYPE(void_to_nothing_t<void*>, void*);
+
+static_assert(is_nothing_v<nothing>);
+static_assert(is_nothing_v<nothing&>);
+static_assert(is_nothing_v<nothing&&>);
+static_assert(is_nothing_v<const nothing>);
+static_assert(is_nothing_v<const nothing&>);
+static_assert(is_nothing_v<const nothing&&>);
+static_assert(!is_nothing_v<nothing*>);
```

Comment:
Since you've decided that a pointer to nothing is not nothing (which I still feel weird about), maybe you should document this in nothing.hpp.